scripts/capture_so101_camera_frame.py

```
@@ -59,24 +74,48 @@ def main() -> int:
 
 
+def _find_repo_root(start: Path) -> Path:
+    for candidate in [start.resolve(), *start.resolve().parents]:
+        if (candidate / "pyproject.toml").is_file() and (candidate / "accountable_swarm").is_dir():
+            return candidate
+    raise ValueError("could not locate repository root")
+
+
+def _repo_path(repo_root: Path, raw_path: Path) -> Path:
+    if raw_path.is_absolute():
+        raise ValueError("artifact paths must be repo-relative")
```

Ruff (0.15.18)

[warning] Avoid specifying long messages outside the exception class

(TRY003)